core_impl/src/main/java/io/opencensus/implcore/tags/propagation/SerializationUtils.java

```diff
@@ -96,27 +96,33 @@ static TagContextImpl deserializeBinary(byte[] bytes) throws TagContextParseExce
         throw new TagContextParseException(
             "Wrong Version ID: " + versionId + ". Currently supported version is: " + VERSION_ID);
       }
-
-      int limit = buffer.limit();
-      while (buffer.position() < limit) {
-        int fieldId = buffer.get();
-        switch (fieldId) {
-          case TAG_FIELD_ID:
-            TagKey key = createTagKey(decodeString(buffer));
-            TagValue val = createTagValue(key, decodeString(buffer));
-            tags.put(key, val);
-            break;
-          default:
-            // TODO(sebright): Skip unknown fields.
-            throw new TagContextParseException("Unsupported tag context field ID: " + fieldId);
-        }
-      }
-      return new TagContextImpl(tags);
+      return new TagContextImpl(parseTags(buffer));
     } catch (BufferUnderflowException exn) {
       throw new TagContextParseException(exn.toString()); // byte array format error.
     }
   }
 
+  private static Map<TagKey, TagValue> parseTags(ByteBuffer buffer)
+      throws TagContextParseException {
+    Map<TagKey, TagValue> tags = new HashMap<TagKey, TagValue>();
+    int limit = buffer.limit();
+    while (buffer.position() < limit) {
+      int type = buffer.get();
+      switch (type) {
```

Comment:
Pease use an if-statement, it is much faster than switch.

Reply:
Done